ui/ui-components/pages/model/[modelId]/property/[propertyId]/edit.tsx

```diff
@@ -66,16 +66,19 @@ export default function Page(props) {
   if (hydrationError) errorHandler.set({ message: hydrationError });
 
   useEffect(() => {
+    setProperty(props.property);
+    setPluginOptions(props.pluginOptions);
+    setLocalFilters(makeLocal(props.property.filters));
     newRuleDefaults();
-  }, []);
+  }, [propertyId]);
 
   useEffect(() => {
     updatePluginOptions();
 
     return () => {
       clearTimeout(timer);
     };
-  }, [JSON.stringify(property.options)]);
+  }, [property.id, JSON.stringify(property.options)]);
```

Comment:
What's the difference between `propertyId` and `property.id`?

Reply:
I think this hook wants to be called after property is set, which the previous hook will do after the URL (propertyID) changes